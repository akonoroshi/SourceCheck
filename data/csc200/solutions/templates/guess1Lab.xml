<project name="guess1Lab" app="iSnap 2.1, http://go.ncsu.edu/isnap" version="1" guid="541caacc-4c0e-4d35-b39a-9b0babafceb9" assignment="guess1Lab"><data>{}</data><notes></notes><thumbnail>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAAB4CAYAAAB1ovlvAAAE1ElEQVR4Xu2dvUtbcRSGz9XQ1iUBQbQQKVIEgwgSBaEOgiB0acCA2C7tv+AsKCiCo2OdS2nBQocmWFzEqTjVDGZLlG6FglI7mCZKSu4ghPhxLYjvOefN4nJvfN7f83CDLgnq9Xpd+OIJ3NMJBAzwnk6evzY8AQbIEO71BC4NcG9vT3Z2du4MLB6PSyaTka6urjv7HXxjHSfQEuDGxoak0+lb0fd0d8vHl08k/rDtxvuevy3Irz+V8Lp8Pi9zc3M33sML7J5AU4C7u7vSeDo1/i4JgiDyz7bzv/L4XUba24JIJ/Xj9deL9x8cHIx0Dy+yeQJNAS4sLEg2mw3jOD09laWlJZmdnZW+vj5JJBLhk3F7e1smJydlZWVFCoWCzM/PS0cskOT7TNMJtT99JkGiR86+f245udKrLxdxDw8P2zxZrop0Ak0Brq2tyfj4eBjH1taWjI6OSiwWC4Ps7OyUkZEROTk5kVwuJ/39/bK4uCjLy8sS73jQEmAsnZX6759yXv7WAlLMfrp4Ao6NjUUC5UU2T6ApwKOjIykWi2EclUpFpqam5Pj4WPb396/9OH4UnEkq9ybyCRVefAjfr1arhU9TvvyeQMsfIevr6+HT7TavxlNyYmIi0i2Hh4dycHAg1WpVhoaGJJlMRrqPF9k8Af4f0KZXNasYoBpVNkEZoE2valYxQDWqbIIyQJte1axigGpU2QRlgDa9qlnFANWosgnKAG16VbOKAapRZROUAdr0qmYVA1SjyiYoA7TpVc0qBqhGlU1QBmjTq5pVDFCNKpugDNCmVzWrGKAaVTZBGaBNr2pWMUA1qmyCMkCbXtWsYoBqVNkEZYA2vapZxQDVqLIJygBtelWzigGqUWUTlAHa9KpmFQNUo8omKAO06VXNKgaoRpVNUAZo06uaVQxQjSqboAzQplc1qxigGlU2QRmgTa9qVjFANapsgjJAm17VrGKAalTZBGWANr2qWcUA1aiyCcoAbXpVs4oBqlFlE5QB2vSqZhUDVKPKJigDtOlVzSoGqEaVTVAGaNOrmlUM8ApVqVQq/Db4mZkZNTI1gjLAa6xls9nwm90br1KpJJubm9Lb26vRMywzA7xBzfT0dMsVQRBIrVaTXC4HK1YLGAP8zwCr1ark83ktnmE5GWDEj+ByuRx+BCeTSViZGsEY4BXWBgYGZHV1VS77CNYoGpWZAaKaccLFAJ2IRp3JAFHNOOFigE5Eo85kgKhmnHAxQCeiUWcyQFQzTrgYoBPRqDMZIKoZJ1wM0Ilo1JkMENWMEy4G6EQ06kwGiGrGCRcDdCIadSYDRDXjhIsBOhGNOpMBoppxwsUAnYhGnckAUc044WKATkSjzmSAqGaccDFAJ6JRZzJAVDNOuBigE9GoMxkgqhknXAzQiWjUmQwQ1YwTLgboRDTqTAaIasYJFwN0Ihp1JgNENeOEiwE6EY06kwGimnHCxQCdiEadyQBRzTjhYoBORKPOZICoZpxwMUAnolFnMkBUM064GKAT0agzGSCqGSdcDNCJaNSZDBDVjBMuBuhENOpMBohqxgkXA3QiGnUmA0Q144SLAToRjTqTAaKaccLFAJ2IRp3JAFHNOOFigE5Eo85kgKhmnHAxQCeiUWcyQFQzTrgYoBPRqDMZIKoZJ1z/AGGEm7c8024cAAAAAElFTkSuQmCC</thumbnail><stage name="Stage" guid="541caacc-4c0e-4d35-b39a-9b0babafceb9" width="480" height="360" costume="0" tempo="60" threadsafe="false" lines="round" codify="false" inheritance="false" sublistIDs="false" scheduled="false" id="1"><pentrails>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAFoCAYAAACPNyggAAAOhUlEQVR4Xu3VwQkAAAjEMN1/abewn7jAQRC64wgQIECAAIF3gX1fNEiAAAECBAiMAHsCAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQIHLFxAWmhEwHPAAAAAElFTkSuQmCC</pentrails><costumes><list id="2"></list></costumes><sounds><list id="3"></list></sounds><variables></variables><blocks></blocks><scripts></scripts><sprites><sprite name="Sprite" idx="1" x="0" y="0" heading="90" scale="1" rotation="1" draggable="true" costume="0" color="80,80,80" pen="tip" id="8"><costumes><list id="9"></list></costumes><sounds><list id="10"></list></sounds><variables></variables><blocks></blocks><scripts><script x="122" y="62"><block id="5077" s="receiveGo"></block><block id="5149" s="doDeclareVariables"><list><l>guess</l></list></block><block id="5151" s="doSetVar"><l>guess</l><block id="5180" s="reportRandom"><l>1</l><l>10</l></block></block><block id="5095" s="doSayFor"><l>Welcome to the guessing game!</l><l>2</l></block><block id="5115" s="doAsk"><l>What&apos;s your name?</l></block><block id="5096" s="doSayFor"><block id="5181" s="reportJoinWords"><list><l>Hello </l><block id="5196" s="getLastAnswer"></block></list></block><l>2</l></block></script><script x="130" y="251"><block id="5182" s="doUntil"><block id="5184" s="reportEquals"><block id="5186" s="getLastAnswer"></block><block id="5185" var="guess"/></block><script><block id="5183" s="doIfElse"><block id="5187" s="reportLessThan"><block id="5188" s="getLastAnswer"></block><block id="5189" var="guess"/></block><script><block id="5193" s="doSayFor"><l>Too low!</l><l>2</l></block></script><script><block id="5194" s="doSayFor"><l>Too high!</l><l>2</l></block></script></block><block id="5192" s="doAsk"><l>Guess a number</l></block></script></block><block id="5195" s="doSayFor"><l>Correct!</l><l>2</l></block></script><script x="445" y="248"><block id="5197" s="doUntil"><block id="5198" s="reportEquals"><block id="5199" s="getLastAnswer"></block><block id="5200" var="guess"/></block><script><block id="5209" s="doAsk"><l>Guess a number</l></block><block id="5201" s="doIfElse"><block id="5202" s="reportEquals"><block id="5203" s="getLastAnswer"></block><block id="5204" var="guess"/></block><script><block id="5205" s="doSayFor"><l>Correct!</l><l>2</l></block></script><script><block id="5212" s="doIfElse"><block id="5213" s="reportLessThan"><block id="5214" s="getLastAnswer"></block><block id="5215" var="guess"/></block><script><block id="5216" s="doSayFor"><l>Too low!</l><l>2</l></block></script><script><block id="5206" s="doSayFor"><l>Too high!</l><l>2</l></block></script></block></script></block></script></block></script><script x="717" y="335"><block id="5240" s="doIf"><block id="5241" s="reportEquals"><block id="5242" s="getLastAnswer"></block><block id="5243" var="guess"/></block><script><block id="5244" s="doSayFor"><l>Correct!</l><l>2</l></block></script></block><block id="5220" s="doIf"><block id="5221" s="reportLessThan"><block id="5222" s="getLastAnswer"></block><block id="5223" var="guess"/></block><script><block id="5224" s="doSayFor"><l>Too low!</l><l>2</l></block></script></block><block id="5228" s="doIf"><block id="5229" s="reportGreaterThan"><block id="5230" s="getLastAnswer"></block><block id="5231" var="guess"/></block><script><block id="5232" s="doSayFor"><l>Too high!</l><l>2</l></block></script></block></script></scripts></sprite><watcher var="guess" style="normal" x="10" y="10" color="243,118,29"/></sprites></stage><hidden></hidden><headers></headers><code></code><blocks></blocks><variables><variable name="guess"><l>0</l></variable></variables><editing></editing></project>