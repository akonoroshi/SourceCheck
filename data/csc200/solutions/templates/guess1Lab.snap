snapshot {
    @defBlock SAY {
        doSayFor(@anything)
    }
    @defBlock ASK {
        doAsk(@anything)
    }
    @+defVar (VAR, GLOBAL, LOCAL) {
        stage {
            sprite {
                script {
                    @+optional {
                        @+or {
                            receiveGo
                            receiveMessage(literal)
                        }
                    }
                    @if (VAR, LOCAL) { doDeclareVariables(list(literal)) }
                    @+defVar (SET_VAR, FIRST, LAST) {
                        @if (SET_VAR, FIRST) { doSetVar(literal, reportRandom(literal, literal)) }
                        @+repeat(1, 2) {
                            @block SAY
                        }
                        @block ASK
                        doSayFor(reportJoinWords(list(literal, getLastAnswer)), literal)
                        @if (SET_VAR, LAST) { doSetVar(literal, reportRandom(literal, literal)) }
                    }
                    @defVar (GUESS_VAR) {
                        @defBlock GUESS {
                            @if (GUESS_VAR) { var }
                            @unless (GUESS_VAR) { getLastAnswer }
                        }
                        @defBlock GET_GUESS {
                            @block ASK
                            @if (GUESS_VAR) { doSetVar(literal, getLastAnswer) }
                        }
                        @defBlock EQ_COMPARE {
                            reportEquals(@block GUESS, var)
                        }
                        @defBlock NOT_EQ_COMPARE {
                            reportLessThan(@or {
                                @inline { @block GUESS, var }
                                @inline { var, @block GUESS }
                            })
                        }

                        @defVar (ASK_TOP) {
                            @defVar (CORRECT_IN_LOOP) {
                                @defBlock EQ_FEEDBACK {
                                    @if (CORRECT_IN_LOOP) {
                                        doIf(@block EQ_COMPARE, script {
                                            @block SAY
                                        })
                                    }
                                }
                                @defBlock NOT_EQ_FEEDBACK_IFS {
                                    @anyOrder {
                                        doIf(reportLessThan(@block GUESS, var), script {
                                            @block SAY
                                            @if (ASK_TOP) { @block ASK }
                                        })
                                        doIf(reportLessThan(var, @block GUESS), script {
                                            @block SAY
                                            @if (ASK_TOP) { @block ASK }
                                        })
                                    }
                                }
                                @defBlock NOT_EQ_FEEDBACK_IF_ELSE {
                                    doIfElse(@block NOT_EQ_COMPARE, script {
                                        @block SAY
                                    }, script {
                                        @block SAY
                                    })
                                    @if (ASK_TOP) { @block ASK }
                                }
                                @defBlock NOT_EQ_FEEDBACK {
                                    @or {
                                        @block NOT_EQ_FEEDBACK_IFS
                                        @block NOT_EQ_FEEDBACK_IF_ELSE
                                    }
                                }

                                @if (ASK_TOP) { @block ASK }
                                doUntil(@block EQ_COMPARE, script {
                                    @unless (ASK_TOP) { @block ASK }
                                    @or {
                                        @inline {
                                            @block EQ_FEEDBACK
                                            @if (ASK_TOP) { @block NOT_EQ_FEEDBACK }
                                            @unless (ASK_TOP) { @block NOT_EQ_FEEDBACK_IFS }
                                        }
                                        doIfElse(@block EQ_COMPARE, script {
                                            @if (CORRECT_IN_LOOP) { doSayFor(literal) }
                                        }, script {
                                            @block NOT_EQ_FEEDBACK
                                        })
                                    }
                                })
                                @unless (CORRECT_IN_LOOP) { doSayFor(literal) }
                            }
                        }

                    }
                }
            }
        }
        @if (VAR, GLOBAL) { var }
    }
}